File: LeetCode/231/main.cpp
class Solution {
public:
    bool isPowerOfTwo(int n) {
        if(n <= 0) {
            return false;
        }
        int mask = (n-1);
        int res = n & (n-1);
        if(res == 0) {
            return true;
        }
        return false;
    }
};

File: LeetCode/231/main.go
func isPowerOfTwo(n int) bool {

    // 8 := 1 0 0 0
    // 7 := 0 1 1 1
    // 8 & 7 = 0 .. so power of 2 ?
    
    return n > 0 && (n & (n - 1)) == 0
}
